kvm/src/lib.rs

```diff
@@ -878,10 +909,35 @@ impl AsRawFd for VcpuFd {
 /// Wrapper for `kvm_cpuid2` which has a zero length array at the end.
 /// Hides the zero length array behind a bounds check.
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-#[derive(Clone, Debug, PartialEq)]
 pub struct CpuId {
-    bytes: Vec<u8>,       // Actually accessed as a kvm_cpuid2 struct.
-    allocated_len: usize, // Number of kvm_cpuid_entry2 structs at the end of kvm_cpuid2.
+    /// Wrapper over kvm_cpuid2 from which we only use the first element.
+    kvm_cpuid: Vec<kvm_cpuid2>,
+    // Number of kvm_cpuid_entry2 structs at the end of kvm_cpuid2.
+    allocated_len: usize,
+}
+
+impl Clone for CpuId {
+    fn clone(&self) -> Self {
+        let mut v = Vec::with_capacity(self.kvm_cpuid.len());
+        for _ in 0..self.kvm_cpuid.len() {
```

Comment:
`vec![kvm_cpuid2::default(); self.kvm_cpuid.len()]` would be nice but I see we're missing an autogenerated `Clone` trait for `kvm_cpuid`. Would it make sense to see about adding it? Several other autogenerated structs have it.

Reply:
Bindgen does not add Clone for structures with incomplete array field. Some info about the reasoning is here: https://github.com/rust-lang/rust-bindgen/issues/1431. I do not want to manually modify autogenerated code because it is terrible to maintain. We can add it for wrappers over structures if needed.